plugins/incoming-packages/manager.py

```
@@ -0,0 +1,576 @@
+        return {}
+
+    def _fetch_image(self, url: str, box: Tuple[int, int]) -> Optional[Image.Image]:
+        try:
+            resp = requests.get(url, headers=self._image_headers(), timeout=8)
+            resp.raise_for_status()
+            img = Image.open(BytesIO(resp.content))
+            img.seek(0)  # first frame of an animated GIF
+            img = img.convert("RGB")
+            img.thumbnail(box, Image.Resampling.LANCZOS)
+            return img
+        except Exception as exc:  # pragma: no cover - network/decoding
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)